iroha/src/lib.rs

```diff
@@ -99,6 +101,7 @@ pub struct Iroha {
     block_sync_message_receiver: Arc<RwLock<BlockSyncMessageReceiver>>,
     world_state_view: Arc<RwLock<WorldStateView>>,
     block_sync: Arc<RwLock<BlockSynchronizer>>,
+    genesis_block: Arc<Option<GenesisBlock>>,
```

Comment:
I think Genesis Block does not need to be wrapped with Arc (Atomic Reference Counter), it is not passed into any threads or async functions, as far as I see.